.github/workflows/test.yml

```diff
@@ -17,15 +17,19 @@ jobs:
     needs: linux-unit-tests
 #    uses: salesforcecli/github-workflows/.github/workflows/unitTestsWindows.yml@main
     uses: ./.github/workflows/tempUnitTestsWindows.yml # temporary replacement
-
-  # Uncomment to enable NUT testing in Github Actions
-  # nuts:
-  #   needs: linux-unit-tests
-  #   uses: salesforcecli/github-workflows/.github/workflows/nut.yml@main
-  #   secrets: inherit
-  #   strategy:
-  #     matrix:
-  #       os: [ubuntu-latest, windows-latest]
-  #     fail-fast: false
-  #   with:
-  #     os: ${{ matrix.os }}
+  e2e:
+    # needs: linux-unit-tests
+    uses: ./.github/workflows/e2e.yml
+    secrets: inherit
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest]
+        command:
+          - 'yarn test:e2e'
+        provider:
+          - 'mcp-provider-dx-core'
```

Comment:
I don't love that this will need to be manually updated every time we add a new tool provider. Is there a way around that?

Reply:
I remember now 😁 

I think this impl. was the original factory implementation you mention:
https://github.com/salesforcecli/mcp/pull/185/files#diff-573f02866e65e9170546bcc63b684526934f1b6196f8bbfaca5312272d217944

the issue is that you can't close the client in tests hooks so there's lot of duplication **and** you also need to `assert.fail` in the catch or the test runner will finish as a success:
https://github.com/salesforcecli/mcp/pull/185/files#diff-573f02866e65e9170546bcc63b684526934f1b6196f8bbfaca5312272d217944